pkg/registry/resource/resourceclaim/strategy.go

```diff
@@ -134,6 +136,7 @@ func (resourceclaimStatusStrategy) PrepareForUpdate(ctx context.Context, obj, ol
 	newClaim.Spec = oldClaim.Spec
 	metav1.ResetObjectMetaForStatus(&newClaim.ObjectMeta, &oldClaim.ObjectMeta)
 
+	dropDeallocatedStatusDevices(newClaim, oldClaim)
```

Comment:
@pohly @johnbelamaric @thockin let me share my thoughts here, I was expecting the drivers to "own" and handle the entire lifecycle of `status.devices[].allocatedDeviceStatus`.

The way is implemented here the apiserver is syncing `status.devices[].allocatedDeviceStatus` with `Status.Allocation.Devices.Results[]` 

> // dropDeallocatedStatusDevices removes the status.devices that were allocated
> // in the oldClaim and that have been removed in the newClaim.

I think this behavior can also cause problems causing hotloops 

1. driver A update the status apiserver deletes its
2. device is deallocated and apiserver removes also the status
3. driver A see status is missing tries to update again but apiserver deletes it again here in strategy.

sounds a bit odd to put on clients the requirement of of watching the Status.Allocation.Devices.Results 

WDYT?

Reply:
Yes, that's correct, I think it would make more sense to remove it when the devide got deallocated. Here is the related conversation in the KEP: https://github.com/kubernetes/enhancements/pull/4861#discussion_r1770979535